docs/Frequently Asked Questions/React Native/index.md

```diff
@@ -0,0 +1,247 @@
+})
+
+self.sdlManager.screenManager.softButtonObjects = [softButton];
+```
+
+### Create the EventEmitter Class
+
+Create the class that will be the listener for the notiification you created above. This class will be sending an receieving messages from your JavaScript code (React Native). There are some required method(s) you must include inorder to be an  `EventEmitter` , these method(s) are in the example below. 
+
+##### Objective-C
+###### SDLEventEmitter.h
+```objc
+#import <React/RCTEventEmitter.h>
+#import <React/RCTBridgeModule.h>
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface SDLEventEmitter : RCTEventEmitter
+
+@end
+
+NS_ASSUME_NONNULL_END
+```
+
+###### SDLEventEmitter.m
+```objc
+#import "SDLEventEmitter.h"
+#import "ProxyManager.h"
+#import <React/RCTConvert.h>
+#import <SmartDeviceLink/SmartDeviceLink.h>
+
+@implementation SDLEventEmitter
+
+RCT_EXPORT_MODULE()
+
+- (instancetype)init {
+    self = [super init];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(getDoActionNotification:) name:<#Notification Name#> object:nil];
+
+    return self;
+}
+
+// Required Method 
+- (NSArray<NSString *> *)supportedEvents {
+    return @[@"DoAction"];
+}
+
+- (void)getDoActionNotification:(NSNotification *)notification {
+    if(self.sdlManager == nil) {
+        self.sdlManager = notification.userInfo[@"sdlManager"];
+    }
+    [self sendEventWithName:@"DoAction" body:@{@"type": @"actionType"}];
+}
+
+@end
+```
+
+##### Swift
+```swift
+@objc(SDLEventEmitter)
+class SDLEventEmitter: RCTEventEmitter {
+
+override init() {
+    NotificationCenter.default.addObserver(self, selector: #selector(doAction(_:)), name: Notification.Name(rawValue: "<#Notification Name#>", object: nil)
+    super.init()
+}
+
+// Required Method 
+@objc func doAction(_ notification: Notification) {
+    if self.sdlManger == nil {
+        self.sdlManager = notification.userInfo["sdlManager"]
+    }
+
+    sendEvent(withName: "DoAction", body: ["type": "actionType"])
+}
+
+override func supportedEvents() -> [String]! {
+    return ["DoAction"]
+}
+
+}
+```
+
+The above example will call into your JavaScript code with an event type `DoAction`. Inside your React Native (JavaScript) code, create an  `NativeEventEmitter` object within your `EventEmitter` module and add a listener for the event.
+
+```javascript
+import { NativeEventEmitter, NativeModules } from 'react-native';
+const  { SDLEventEmitter } = NativeModules;
+
+const testEventEmitter = new NativeEventEmitter(SDLEventEmitter);
+
+// Build a listener to listen for events
+const testData = testEventEmitter.addListener(
+    'DoAction',
+        () => SDLEventEmitter.eventCall({
+            "data": {
+                "low": "77",
+                "high": "87",
+                "currentTemp": "82",
+                "rain": "50%"
+            }
+        }   
+    )
+)
+```
+### Exposing Methods
```

Comment:
```suggestion

### Exposing Methods
```

Reply:
Exposing methods to which side? `Exposing JavaScript Methods to ProxyManager`? That doesn't seem to be what's going on below. What does it mean to expose a method? Why would you want to do it?